appium/webdriver/webdriver.py

```diff
@@ -204,28 +205,21 @@ class WebDriver(
 ):
     def __init__(  # noqa: PLR0913
         self,
-        command_executor: Union[str, AppiumConnection] = 'http://127.0.0.1:4444/wd/hub',
-        keep_alive: bool = True,
-        direct_connection: bool = True,
+        command_executor: Union[str, AppiumConnection] = 'http://127.0.0.1:4723',
         extensions: Optional[List['WebDriver']] = None,
-        strict_ssl: bool = True,
+        file_detector: Optional[FileDetector] = None,
         options: Union[AppiumOptions, List[AppiumOptions], None] = None,
-        client_config: Optional[ClientConfig] = None,
+        client_config: Optional[AppiumClientConfig] = None,
     ):
         if isinstance(command_executor, str):
```

Comment:
would it be possible to extract this logic into a separate method, so super class init would be the very first call there?

Reply:
> super class init would be the very first call there?

Actually we could by modifying here like below:
```
    def __init__(
        self,
        command_executor: AppiumConnection,
        client_config: AppiumClientConfig,
        extensions: Optional[List['WebDriver']] = None,
        file_detector: Optional[FileDetector] = None,
        options: Union[AppiumOptions, List[AppiumOptions], None] = None,
    ):
```

Then, a new method will help to build client_config and command_executor.

To avoid handling string of `command_executor` in selenium, which uses `ClientConfig`,  I have added these lines to keep using `AppiumClientConfig` as appium python client for such a string case (to keep providing similar syntax to webdriver.Remote for selenium and appium)